lib/active_merchant/billing/gateways/hps.rb

```diff
@@ -254,32 +253,24 @@ def add_descriptor_name(xml, options)
         xml.hps :TxnDescriptor, options[:descriptor_name] if options[:descriptor_name]
       end
 
-      def add_three_d_secure(xml, card_or_token, options)
-        if card_or_token.is_a?(NetworkTokenizationCreditCard)
-          build_three_d_secure(xml, {
-            source: card_or_token.source,
-            cavv: card_or_token.payment_cryptogram,
-            eci: card_or_token.eci,
-            xid: card_or_token.transaction_id
-          })
-        elsif options[:three_d_secure]
-          options[:three_d_secure][:source] ||= card_brand(card_or_token)
-          build_three_d_secure(xml, options[:three_d_secure])
+      def add_wallet_data(xml, payment_method, options)
+        return unless payment_method.is_a?(NetworkTokenizationCreditCard)
+
+        xml.hps :WalletData do
+          xml.hps :PaymentSource, PAYMENT_DATA_SOURCE_MAPPING[payment_method.source]
+          xml.hps :Cryptogram, payment_method.payment_cryptogram
+          xml.hps :ECI, payment_method.eci&.to_s&.sub!(/^0/, '') if payment_method.eci
```

Comment:
Why not use `strip_leading_zero`?

Reply:
I can't even remember why but I'll update it do use strip_leading_zero